### PR TITLE
Added asHexString() in AbstractByteArrayAssert

### LINKED ISSUE #1839: ByteArrayAssert with Hex-Strings
#### Summary
Hi i am using a lot of hexadecimal strings in my byte[] Tests. I think the code and the dependency tree could be nicer.
What do you think about this method?

I will write a pull request if this improvement is welcome. 

#### Example
Usage of the new method:
```java
 assertThat(new byte[] { (byte) 255 , 2  }).asHexString().isEqualTo("FF02");

 assertThat(new byte[] { (byte) 255 , 2  }).asHexString().isEqualTo("AF02");
org.opentest4j.AssertionFailedError: 
Expecting:
 <"FF02">
to be equal to:
 <"AF02">
but was not.
```
Implementation suggestion:
```java
public abstract class AbstractByteArrayAssert<SELF extends AbstractByteArrayAssert<SELF>>
    extends AbstractArrayAssert<SELF, byte[], Byte> {
....
  public AbstractStringAssert<?> asHexString() {
    objects.assertNotNull(info, actual);
    return Assertions.assertThat(Hexadecimals.toHexString(actual));
  }
```

### PR DESCRIPTION
**Pull request by AndreasMager**
#### Check List:
* Fixes #1839
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/util/Hexadecimals.java [hunk 1]
```diff
@@ -17,15 +17,32 @@
  */
 public class Hexadecimals {
 
+  private static final byte[] charToByte = new byte[256];
```
[the remark this answers is not in the document]
Comment (AndreasMager) [reply src/main/java/org/assertj/core/util/Hexadecimals.java]: Sorry left over from 'fromHexString'.

**Comment by AndreasMager** [comment]
Sorry for the errors yesterday was a long day.
Thanks @joel-costigliola

**Comment by joel-costigliola** [comment]
No problem @AndreasMager

**Comment by joel-costigliola** [comment]
Integrated thanks @AndreasMager!

**Pull request closed without merging**